### PR TITLE
Add device permissions and device resource to documentation

### PR DESCRIPTION
**Pull request by tigerli-okta**
## Description:
https://64d15fb7a20ba1305ef5afb9--reverent-murdock-829d24.netlify.app
- Role Assignments page permission types: https://64d15fb7a20ba1305ef5afb9--reverent-murdock-829d24.netlify.app/docs/concepts/role-assignment/#permission-types
- Role Assignments page supported resources: https://64d15fb7a20ba1305ef5afb9--reverent-murdock-829d24.netlify.app/docs/concepts/role-assignment/#supported-resources
- Roles API permission types: https://64d15fb7a20ba1305ef5afb9--reverent-murdock-829d24.netlify.app/docs/reference/api/roles/#permission-properties

- Added device permissions and device resources to Role Assignment and Administrator Roles page
- **Is this PR related to a Monolith release?** Yes (2023.08.1 weekly)

### Resolves:

* [OKTA-619366](https://oktainc.atlassian.net/browse/OKTA-619366)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md [hunk 1]
```diff
@@ -245,5 +246,13 @@ Standard resource sets are managed by Okta only and can't be updated or deleted.
...
+| `okta.devices.lifecycle.delete` <br><ApiLifecycle access="ea" />                              | Allows the admin to permanently delete devices                                                                                                 | All Devices
+| `okta.devices.read` <br><ApiLifecycle access="ea" />                              | Allows the admin to read device details and information                                                                                                 | All Devices
 > **Note:** Governance permissions are currently only supported as part of the [Standard IAM-based Roles](#iam-based-standard-role-types). You can't use these to create or update other roles.
 > **Note:** `okta.apps.manageFirstPartyApps` permission is currently only supported as part of some [Standard IAM-based Roles](/docs/concepts/role-assignment/#iam-based-standard-role-types). You can't use it to create or update other roles.
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:258]:
```suggestion

> **Note:** `okta.apps.manageFirstPartyApps` permission is currently only supported as part of some [Standard IAM-based Roles](/docs/concepts/role-assignment/#iam-based-standard-role-types). You can't use it to create or update other roles.

> **Note:** `okta.devices.*` permissions are self-service Early Access. Turn on the **Enable custom admin roles for device permissions** feature from the **Settings** > **Feature** page in the Admin Console to access these permissions. See [Manage Early Access and Beta features](https://help.okta.com/okta_help.htm?id=ext_Manage_Early_Access_features).
```

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md [hunk 2]
```diff
@@ -245,5 +246,13 @@ Standard resource sets are managed by Okta only and can't be updated or deleted.
...
+| `okta.devices.lifecycle.suspend` <br><ApiLifecycle access="ea" />                             | Allows the admin to suspend device access to Okta                                      | All Devices
+| `okta.devices.lifecycle.unsuspend` <br><ApiLifecycle access="ea" />                           | Allows the admin to unsuspend and restore device access to Okta                                                                                                      | All Devices
+| `okta.devices.lifecycle.delete` <br><ApiLifecycle access="ea" />                              | Allows the admin to permanently delete devices                                                                                                 | All Devices
+| `okta.devices.read` <br><ApiLifecycle access="ea" />                              | Allows the admin to read device details and information                                                                                                 | All Devices
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:256]:
```suggestion
| `okta.devices.read` <br><ApiLifecycle access="ea" />                              | Allows the admin to read device details                                                                                                 | All Devices
```

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md [hunk 3]
```diff
@@ -245,5 +246,13 @@ Standard resource sets are managed by Okta only and can't be updated or deleted.
...
+| `okta.devices.manage`  <br><ApiLifecycle access="ea" />  | Allows the admin to manage devices and perform all device lifecycle operations                                                                                       | All Devices |
+| `okta.devices.lifecycle.manage`  <br><ApiLifecycle access="ea" />  | Allows the admin to perform any device lifecycle operations                                                                                    | All Devices |
+| `okta.devices.lifecycle.activate` <br><ApiLifecycle access="ea" />                            | Allows the admin to activate devices                                                                                                           | All Devices
+| `okta.devices.lifecycle.deactivate` <br><ApiLifecycle access="ea" />                          | Allows the admin to deactivate devices. Deactivation causes a device to lose all device user links.                                                                                                          | All Devices
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:252]:
```suggestion
| `okta.devices.lifecycle.deactivate` <br><ApiLifecycle access="ea" />                          | Allows the admin to deactivate devices. When a device is deactivated, it loses all device user links.                                                                                                          | All Devices
```

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md [hunk 4]
```diff
@@ -146,6 +146,7 @@ The following are the supported resources.
...
 |                         | A specific Group                                                    | `orn:${partition}:directory:${yourOrgId}:groups:${groupId}`                           | [`https://${yourOktaDomain}/api/v1/groups/${groupId}`](/docs/reference/api/groups/#get-group)                                                           |
 |                         | All Users within a specific Group                                   | `orn:${partition}:directory:${yourOrgId}:groups:${groupId}:contained_resources`       | [`https://${yourOktaDomain}/api/v1/groups/${groupId}/users`](/docs/reference/api/groups/#list-group-members)                                            |
+|                         | All Devices     <br><ApiLifecycle access="ea" />                                                     | `orn:${partition}:directory:${yourOrgId}:devices`                                     | [`https://${yourOktaDomain}/api/v1/devices`](/docs/reference/api/devices)                                            |
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:149]: Can you please add this note at the bottom of this table?

> **Note:** The All Devices resource is self-service Early Access. Turn on the **Enable custom admin roles for device permissions** feature from the **Settings** > **Feature** page in the Admin Console to use this resource. See [Manage Early Access and Beta features](https://help.okta.com/okta_help.htm?id=ext_Manage_Early_Access_features).

File: packages/@okta/vuepress-site/docs/reference/api/roles/index.md [hunk 5]
```diff
@@ -4137,6 +4145,7 @@ The following are the supported resources.
...
 |                         | A specific Group                                                    | `orn:${partition}:directory:${yourOrgId}:groups:${groupId}`                           | [`https://${yourOktaDomain}/api/v1/groups/${groupId}`](/docs/reference/api/groups/#get-group)                                                           |
 |                         | All Users within a specific Group                                   | `orn:${partition}:directory:${yourOrgId}:groups:${groupId}:contained_resources`       | [`https://${yourOktaDomain}/api/v1/groups/${groupId}/users`](/docs/reference/api/groups/#list-group-members)                                            |
+|                         | All Devices     <br><ApiLifecycle access="ea" />                                                     | `orn:${partition}:directory:${yourOrgId}:devices`                                     | [`https://${yourOktaDomain}/api/v1/devices`](/docs/reference/api/devices)
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/api/roles/index.md:4148]: Can you please add this note at the bottom of this table?

> **Note:** The All Devices resource is self-service Early Access. Turn on the **Enable custom admin roles for device permissions** feature from the **Settings** > **Feature** page in the Admin Console to use this resource. See [Manage Early Access and Beta features](https://help.okta.com/okta_help.htm?id=ext_Manage_Early_Access_features).

File: packages/@okta/vuepress-site/docs/reference/api/roles/index.md [hunk 6]
```diff
@@ -4085,6 +4085,14 @@ User permissions are only effective regarding the Groups to which the admin is g
...
+| `okta.devices.lifecycle.suspend` <br><ApiLifecycle access="ea" />                             | Allows the admin to suspend device access to Okta                                      | All Devices
+| `okta.devices.lifecycle.unsuspend` <br><ApiLifecycle access="ea" />                           | Allows the admin to unsuspend and restore device access to Okta                                                                                                      | All Devices
+| `okta.devices.lifecycle.delete` <br><ApiLifecycle access="ea" />                              | Allows the admin to permanently delete devices                                                                                                 | All Devices
+| `okta.devices.read` <br><ApiLifecycle access="ea" />                              | Allows the admin to read device details and information                                                                                                 | All Devices
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/api/roles/index.md:4095]:
```suggestion
| `okta.devices.read` <br><ApiLifecycle access="ea" />                              | Allows the admin to read device details                                                                                                 | All Devices
```

File: packages/@okta/vuepress-site/docs/reference/api/roles/index.md [hunk 7]
```diff
@@ -4085,6 +4085,14 @@ User permissions are only effective regarding the Groups to which the admin is g
...
+| `okta.devices.manage`  <br><ApiLifecycle access="ea" />  | Allows the admin to manage devices and perform all device lifecycle operations                                                                                       | All Devices |
+| `okta.devices.lifecycle.manage`  <br><ApiLifecycle access="ea" />  | Allows the admin to perform any device lifecycle operations                                                                                    | All Devices |
+| `okta.devices.lifecycle.activate` <br><ApiLifecycle access="ea" />                            | Allows the admin to activate devices                                                                                                           | All Devices
+| `okta.devices.lifecycle.deactivate` <br><ApiLifecycle access="ea" />                          | Allows the admin to deactivate devices. Deactivation causes a device to lose all device user links.                                                                                                          | All Devices
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/api/roles/index.md:4091]:
```suggestion
| `okta.devices.lifecycle.deactivate` <br><ApiLifecycle access="ea" />                          | Allows the admin to deactivate devices. When a device is deactivated, it loses all device user links.                                                                                                          | All Devices
```

File: packages/@okta/vuepress-site/docs/reference/api/roles/index.md [hunk 8]
```diff
@@ -4085,6 +4085,14 @@ User permissions are only effective regarding the Groups to which the admin is g
...
+| `okta.devices.lifecycle.delete` <br><ApiLifecycle access="ea" />                              | Allows the admin to permanently delete devices                                                                                                 | All Devices
+| `okta.devices.read` <br><ApiLifecycle access="ea" />                              | Allows the admin to read device details and information                                                                                                 | All Devices
 > **Note:** Governance permissions are currently only supported as part of the [Standard IAM-based Roles](/docs/concepts/role-assignment/#iam-based-standard-role-types). You can't use these to create or update other roles.
 > **Note:** `okta.apps.manageFirstPartyApps` permission is only supported as part of some [Standard IAM-based Roles](/docs/concepts/role-assignment/#iam-based-standard-role-types). You can't use it to create or update other roles.
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/api/roles/index.md:4097]:
```suggestion

> **Note:** `okta.apps.manageFirstPartyApps` permission is only supported as part of some [Standard IAM-based Roles](/docs/concepts/role-assignment/#iam-based-standard-role-types). You can't use it to create or update other roles.

> **Note:** `okta.devices.*` permissions are self-service Early Access. Turn on the **Enable custom admin roles for device permissions** feature from the **Settings** > **Feature** page in the Admin Console to access these permissions. See [Manage Early Access and Beta features](https://help.okta.com/okta_help.htm?id=ext_Manage_Early_Access_features).
```

**Review by vanngo-okta — CHANGES_REQUESTED** [review]
Hi @tigerli-okta, your edits look good! I only have a few nits and can you please add the EA note at the bottom of the tables? Thank you!

**Review by brentschaus-okta — APPROVED** [review]
LGTM!

**Review by vanngo-okta — APPROVED** [review]
LGTM! 🚀

**Pull request merged**